content/docs/guides/ha_scale/perf-scale.md

```diff
@@ -0,0 +1,87 @@
+
+<p align="center">
+  <img src="/docs/images/perf/mesh-topo.png" width="650"/>
+</p>
+
+During the test, the echo applications will be deployed all at the same time. Therefore, the control plane receives a load spike after the deployment starts. All the Envoy sidecars are connecting to the control plane simutaneously. After that, the control plane becomes much more idle. We deliberately create this load spike at test bootstrap time in order to explore the maximum scale that the mesh can handle. Each round of test will last for 15 minutes.
+
+## Performance
+
+In this test, we focus on the latency overhead and resource consumption added by OSM's. We currently test the per pod RPS at 100, 500, 1000, 2000 and 4000 respectively.
+
+### Latency
+
+Below is the request latency at different RPS levels with and without OSM installed. Note, with OSM installed the added latency comes from both the client and server sidecar.
+
+<p align="center">
+  <img src="/docs/images/perf/wo-osm-latency.png" width="650"/>
+</p>
+
+<p align="center">
+  <img src="/docs/images/perf/w-osm-latency.png" width="650"/>
+</p>
+
+The application without OSM performs quite similar across different RPS levels, although there is an upward trend on the 99.9 percentile latency. Whereas when OSM is installed, the latency below 1000 RPS does not change much. It starts to increase proportially when latency is above 1000 RPS. Looking at the overhead that OSM adds to the P99 latency, it is less than 2ms for RPS below 2000 and around 14ms and 30ms at 2000 and 4000 RPS respectively.
+
+Note that our test setup is very simple, with a single client/server relation. Your mileage may vary with more complex applications, or with different traffic policies set on the mesh.
+
+### Resource Consumption
+
+Control plane resource consumption is stable as it is not relavant to the data plane request rate.
+
+For the sidecars on the data plane, the memory usage is relatively consistent across different RPS levels, which is around 140MB. The CPU usage has a clear increasing trend and it is nearly proportional to the RPS level. In our case, roughly every 100 RPS takes 0.1 cores of CPU on sidecar container.
+
+<p align="center">
+  <img src="/docs/images/perf/avg-cpu.png" width="650"/>
+</p>
+
+## Scalability
+
+In the scalability test, we explore how resource consumption changes for both the control plane and the data plane as the mesh scales up. A number of echo services will be added to the mesh. Each service will have 10 replicas. The mesh is eventually added 200, 400, 800 and 1600 pods respectively in each round of the test. The load generator service is not used in scalability test.
+
+### Control Plane
+
+#### CPU Usage
+
+Below shows the CPU usage of control plane at both pod-churn time and stable time:
```

Comment:
Do pod-churn, peak, and spike all have the same meaning in this doc?

Reply:
Good question. Yes they are the same thing. I will make them consistent.